docker/centos/Dockerfile

```diff
@@ -0,0 +1,98 @@
+# This Dockerfile enables an iterative development workflow where you can make
+# a change and test it out quickly. The majority of commands in this file will
+# be cached, making the feedback loop typically quite short. The workflow is
+# as follows:
+#   1. Set up pre-conditions for the system in puppet code using `deploy.pp`.
+#   2. Make a change to the module.
+#   3. Run `docker build -f docker/Dockerfile .` or `./docker/bin/run-centos.sh`
+#      from the project directory. If you would like to test specific version
+#      updates, you can add run this like so:
+#        `docker build -f docker/centos/Dockerfile . \
+#           -t pa-dev:centos --build-arg before=1.10.14`
+#   4. Upgrade the container by running the image:
+#        `docker run -it pa-dev:centos`
+#      Specify your upgrade TO version as an argument to the `docker run`
+#      command.
+#   5. Review the output. Repeat steps 2-5 as needed.
+#
+# At the end of execution, you will see a line like:
+#
+# Notice: /Stage[main]/Puppet_agent::Install/Package[puppet-agent]/ensure: ensure changed '1.10.14-1.el7' to '6.2.0'
+#
+# This specifies the versions that were used for upgrade.
+#
+# Arguments:
+# - before: The version to do upgrade FROM. Default: "1.10.14"
+
+FROM centos:7
+
+# Use this to force a cache reset (e.g. for output purposes)
+#COPY docker/Dockerfile /tmp/Dockerfile
+
+# Install some other dependencies for ease of life.
+RUN  yum update -y \
+  && yum install -y wget git \
+  && yum clean all
+
+# Install several repos: PC1 (puppet 4), puppet 5, and puppet 6.
+RUN wget -O puppet-pc1.rpm http://yum.puppetlabs.com/puppetlabs-release-pc1-el-7.noarch.rpm && \
+    rpm -i puppet-pc1.rpm --force --replacefiles && \
+    wget -O puppet5.rpm http://yum.puppetlabs.com/puppet5/puppet5-release-el-7.noarch.rpm && \
```

Comment:
One of the tasks the module performs is installing these repos. Can we parameterize which repo gets installed so the development workflow can also test out the "install the repo" functionality?